app/lib/actions/pulp/orchestration/repository/sync.rb

```diff
@@ -0,0 +1,22 @@
+module Actions
+  module Pulp
+    module Orchestration
+      module Repository
+        class Sync < Pulp::Abstract
+          def plan(_repository, _smart_proxy, options)
+            sequence do
+              action_output = plan_action(Actions::Pulp::Repository::Sync, options).output
+              plan_self(:output => action_output)
+            end
+          end
+
+          def run
+            input[:output].each do |key, value|
+              output[key] = value
+            end
```

Comment:
since this is repeated, i wonder if we could turn this into middleware, similar to: https://github.com/Katello/katello/blob/master/app/lib/actions/middleware/skip_if_matching_content.rb#L4-L6